src/consensus/vote.hpp

```diff
@@ -191,9 +259,12 @@ class VoteManager {
   using upgradableLock_ = boost::upgrade_lock<boost::shared_mutex>;
   using upgradeLock_ = boost::upgrade_to_unique_lock<boost::shared_mutex>;
 
-  // <pbft_round, <vote_hash, vote>>
+  // <pbft round, <vote hash, vote>>
   std::map<uint64_t, std::unordered_map<vote_hash_t, Vote>> unverified_votes_;
-  std::map<uint64_t, std::unordered_map<vote_hash_t, Vote>> verified_votes_;
+
+  // <PBFT round, <PBFT step, <voted value, <vote hash, vote>>>>
+  std::map<uint64_t, std::map<size_t, std::unordered_map<blk_hash_t, std::unordered_map<vote_hash_t, Vote>>>>
```

Comment:
do we need to have this steps and round in order? we are accessing it by step/round, so I would suggest  unordered_map for performance. Even for `unverified_votes_`

Reply:
The round and step need in order.
For round, if a bootstrapping node join. We need to check the round in descending to find the current round in the network. If we can find enough votes in round 1000, then we could jump to round 1000 directly.
For step, we need to check in ascending to check enough votes to go to next round. For example, like node0 at step 5 and node1 at step 100. So we will get enough votes at step 5 because node0 hasn't vote in the further step